src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -973,7 +973,7 @@ public override async Task WaitForCancellationAsync(bool ignoreIncomingData = tr
                 {
                     Assert.Equal(FrameType.RstStream, frame.Type);
                 }
-            } while (frame.Type != FrameType.RstStream);
+            } while (frame.Type != FrameType.RstStream || (frame is RstStreamFrame rstStreamFrame && rstStreamFrame.ErrorCode != 0x8));
```

Comment:
```suggestion
            } while (frame.Type != FrameType.RstStream || (frame is RstStreamFrame rstStreamFrame && rstStreamFrame.ErrorCode != ProtocolErrors.CANCEL));
```
I  think

Reply:
ErrorCode property is an int, so I casted the enum